src/graphql.ts

```diff
@@ -225,12 +225,10 @@ directive @computed on FIELD_DEFINITION
         if (ref) {
           let target = expect(ref.typenames[0], types);
           let relationship = expect(ref.key, relationships);
+          let targetPreset = Array.isArray(value) ? value.map(content => transform(target, content)) : transform(target, value as Record<string, unknown>);
```

Comment:
I'm going to approve, but I think we should have a generic type parameter for `Preset` to stop all the type assertions and another thing I would like to try is to stop all the defaults like `GraphGen<API = Record<string, any>`

I don't think we need many of these defaults and it might make the inference a little worse

Reply:
We do currently have a type parameter to `Preset` which is basically like partial, except it is recursive. In this case, it was incorrectly transforming to pass down to the the graph layer which is totally untyped. The outer, SDL layer does enforce preset types correctly (I think) 

But any way you see to improve the typing, let's go for it 👍🏻